services/rule_engine/src/main/java/agents/Smith.java

```diff
@@ -82,12 +83,22 @@ public void createNewAgents(Spreadsheet spreadsheet) {
         List<Map<String, String>> mailList = spreadsheet.getSheets().get("Mailovi");
         nextEmailRow = mailList.size() + 2;
 
+        List<String> mailovi = new ArrayList<String>();
+
         for (Map<String, String> item : mailList) {
             for (String key : item.keySet()) {
                 String email = item.get(key);
                 if(!Agents.exists(email)){
                     Agents.createAgent(new Moli(email));
                 }
+                mailovi.add(email);
+            }
+        }
+
+        List<Moli> agents = Agents.getByType(Moli.class);
+        for (Moli m : agents) {
+            if (!mailovi.contains(m.getId())) {
```

Comment:
This will also delete Moli agents for people that are still in the "Enter your email" flow, and we don't want that to happen until the flow finished. We should add a check here that the agent had an email connection set and that the email connection is not in the list of active emails (don't use the agent's id for this).